propagate_const.h

```diff
@@ -41,68 +41,68 @@ class propagate_const {
 
  public:
   // [propagate_const.ctor], constructors
-  constexpr propagate_const() = default;
```

Comment:
Prefer to use some CONSTEXPR macro instead. We don't want to limit constexpr for all the compilers, isn't it?

Reply:
Agreed. I won’t merge this PR in its current state.